pulstar/src/utils.rs

```diff
@@ -0,0 +1,76 @@
+//use std::any::Any;
+//use std::hint::assert_unchecked;
+use temp_name_lib::type_def::{Eigenfunctions,Config};
+#[derive(Debug,PartialEq)]
+pub struct PulstarConfig{
+    pub time_pts_nmbr:u16,
+    pub mode_config:Config,
+    pub freqcycli:Vec<f64>,
+    pub temperature_config:Vec<Eigenfunctions>,
+    pub gravity_config:Vec<Eigenfunctions>,
+    pub star_config:StarInfo,
+    pub is_time_dependent:bool,
+    pub suppress_pulse:bool,
+    pub print_amplitude:bool,
+}
+#[derive(Debug,PartialEq)]
+pub struct StarInfo{
+    pub mass:f64,
+    pub radius:f64,
+    pub effective_temperature:f64,
+    pub rotation_velocity:f64,
+    pub inclination_angle:i16,
+}
+
+mod parse_value;
+mod parse_string;
+pub mod parse_file;
+
+pub mod write_grid_data;
+
+pub mod print_info;
+
+#[derive(Debug,Clone,Copy)]
+enum InputKind{
+    F64(f64),
+    U16(u16),
+    I16(i16),
+    BOOL(bool),
+}
+
+enum InputLines{
```

Comment:
All of this Un-Rusty code can be avoided if you convert the inputfile to YAML format.